.circleci/setup.sh

```diff
@@ -28,6 +28,9 @@ if [[ ! -d $WORKSPACE/miniconda ]]; then
     bash miniconda.sh -b -p $WORKSPACE/miniconda
 
     # step 2: setup channels
+
+    # Added to support the new compiler packages from Anaconda 5
+    conda config --system --add channels anaconda
```

Comment:
I think we can remove the default channel here. anaconda is default + extra afaik or even a symlink.

Reply:
We should **not** use the `anaconda` channel. That channel should not be used under normal circumstances. IIRC, this is just a mirror (or intermediate source? -- can't remember) of `defaults` (or rather the `pkgs/main` part, I think).
`defaults` has everything we need, including the compilers.

Reply:
The default compilers used by `conda-build 3` should be packages from:

<details><summary>

```
$ for s in linux-64 osx-64 win-64; do conda search "*_${s}[subdir=${s}]" ; done
```

</summary>

```                                                                                                                             
Loading channels: done
# Name                  Version           Build  Channel             
binutils_impl_linux-64          2.28.1      h04c84fa_2  pkgs/main           
binutils_impl_linux-64          2.28.1      had2808c_3  pkgs/main           
binutils_linux-64          2.28.1      he4fe6c7_1  pkgs/main           
binutils_linux-64           5.4.0     had2808c_24  pkgs/main           
binutils_linux-64           7.2.0              19  pkgs/main           
binutils_linux-64           7.2.0              24  pkgs/main           
binutils_linux-64           7.2.0              25  pkgs/main           
binutils_linux-64           7.2.0              26  pkgs/main           
binutils_linux-64           7.2.0     hc67e822_10  pkgs/main           
binutils_linux-64           7.2.0     hc67e822_11  pkgs/main           
binutils_linux-64           7.2.0     hc67e822_12  pkgs/main           
binutils_linux-64           7.2.0     hc67e822_13  pkgs/main           
binutils_linux-64           7.2.0     hc67e822_14  pkgs/main           
binutils_linux-64           7.2.0     hc67e822_15  pkgs/main           
binutils_linux-64           7.2.0      hc67e822_2  pkgs/main           
binutils_linux-64           7.2.0      hc67e822_3  pkgs/main           
binutils_linux-64           7.2.0      hc67e822_7  pkgs/main           
binutils_linux-64           7.2.0      hc67e822_8  pkgs/main           
binutils_linux-64           7.2.0      hc67e822_9  pkgs/main           
duma_linux-64            2.5.15      h28ad9a3_3  pkgs/main           
duma_linux-64            2.5.15      h49555fe_2  pkgs/main           
duma_linux-64            2.5.15      hec2b5b5_1  pkgs/main           
gcc-dbg_linux-64           7.2.0              19  pkgs/main           
gcc-dbg_linux-64           7.2.0              24  pkgs/main           
gcc-dbg_linux-64           7.2.0              25  pkgs/main           
gcc_impl_linux-64           5.4.0      habb00fd_3  pkgs/main           
gcc_impl_linux-64           7.2.0      habb00fd_3  pkgs/main           
gcc_impl_linux-64           7.2.0      hc5ce805_2  pkgs/main           
gcc_linux-64              5.4.0     h98af8de_24  pkgs/main           
gcc_linux-64              7.2.0              19  pkgs/main           
gcc_linux-64              7.2.0              24  pkgs/main           
gcc_linux-64              7.2.0              25  pkgs/main           
gcc_linux-64              7.2.0              26  pkgs/main           
gcc_linux-64              7.2.0     h196ecd4_10  pkgs/main           
gcc_linux-64              7.2.0     h217957b_13  pkgs/main           
gcc_linux-64              7.2.0      h22f992b_7  pkgs/main           
gcc_linux-64              7.2.0      h29fd5c9_3  pkgs/main           
gcc_linux-64              7.2.0     h4118e58_11  pkgs/main           
gcc_linux-64              7.2.0      h60973fd_9  pkgs/main           
gcc_linux-64              7.2.0      h6f34251_8  pkgs/main           
gcc_linux-64              7.2.0     haf1f6fa_15  pkgs/main           
gcc_linux-64              7.2.0      hc7b1ceb_1  pkgs/main           
gcc_linux-64              7.2.0     hd763dfe_12  pkgs/main           
gcc_linux-64              7.2.0      hecb3f9c_2  pkgs/main           
gcc_linux-64              7.2.0     hf1c97a4_14  pkgs/main           
gdb_linux-64             7.12.1      h0d70bb3_3  pkgs/main           
gdb_linux-64             7.12.1      hd7315ee_1  pkgs/main           
gdb_linux-64             7.12.1      hf9d566d_2  pkgs/main           
gfortran-dbg_linux-64           7.2.0              19  pkgs/main           
gfortran-dbg_linux-64           7.2.0              24  pkgs/main           
gfortran-dbg_linux-64           7.2.0              25  pkgs/main           
gfortran_impl_linux-64           5.4.0      hdf63c60_3  pkgs/main           
gfortran_impl_linux-64           7.2.0      hb3c8cce_2  pkgs/main           
gfortran_impl_linux-64           7.2.0      hdf63c60_3  pkgs/main           
gfortran_linux-64           5.4.0     h98af8de_24  pkgs/main           
gfortran_linux-64           7.2.0              19  pkgs/main           
gfortran_linux-64           7.2.0              24  pkgs/main           
gfortran_linux-64           7.2.0              25  pkgs/main           
gfortran_linux-64           7.2.0              26  pkgs/main           
gfortran_linux-64           7.2.0      h0d86fa5_2  pkgs/main           
gfortran_linux-64           7.2.0      h36698f7_3  pkgs/main           
gfortran_linux-64           7.2.0     h3a66261_14  pkgs/main           
gfortran_linux-64           7.2.0      h6c08c0b_8  pkgs/main           
gfortran_linux-64           7.2.0      h6fbc3aa_9  pkgs/main           
gfortran_linux-64           7.2.0      h939f3e1_1  pkgs/main           
gfortran_linux-64           7.2.0     h9b3fb79_10  pkgs/main           
gfortran_linux-64           7.2.0     ha0c5ce7_12  pkgs/main           
gfortran_linux-64           7.2.0     hbff877f_13  pkgs/main           
gfortran_linux-64           7.2.0      hc3a7fdf_7  pkgs/main           
gfortran_linux-64           7.2.0     hc6796cf_11  pkgs/main           
gfortran_linux-64           7.2.0     hd729bf5_15  pkgs/main           
gxx-dbg_linux-64           7.2.0              19  pkgs/main           
gxx-dbg_linux-64           7.2.0              24  pkgs/main           
gxx-dbg_linux-64           7.2.0              25  pkgs/main           
gxx_impl_linux-64           5.4.0      hdf63c60_3  pkgs/main           
gxx_impl_linux-64           7.2.0      hd3faf3d_2  pkgs/main           
gxx_impl_linux-64           7.2.0      hdf63c60_3  pkgs/main           
gxx_linux-64              5.4.0     h98af8de_24  pkgs/main           
gxx_linux-64              7.2.0              19  pkgs/main           
gxx_linux-64              7.2.0              24  pkgs/main           
gxx_linux-64              7.2.0              25  pkgs/main           
gxx_linux-64              7.2.0              26  pkgs/main           
gxx_linux-64              7.2.0     h2e50c1c_15  pkgs/main           
gxx_linux-64              7.2.0      h33f4dac_3  pkgs/main           
gxx_linux-64              7.2.0      h3aab023_7  pkgs/main           
gxx_linux-64              7.2.0     h41d9e3f_12  pkgs/main           
gxx_linux-64              7.2.0     h4bc98f9_14  pkgs/main           
gxx_linux-64              7.2.0      h57110d4_9  pkgs/main           
gxx_linux-64              7.2.0     h5e79354_11  pkgs/main           
gxx_linux-64              7.2.0      h60a51f1_2  pkgs/main           
gxx_linux-64              7.2.0     h7798672_13  pkgs/main           
gxx_linux-64              7.2.0      h8cc0f3c_1  pkgs/main           
gxx_linux-64              7.2.0     hc01b44c_10  pkgs/main           
gxx_linux-64              7.2.0      he93fcd3_8  pkgs/main           
Loading channels: done
# Name                  Version           Build  Channel             
clang-dbg_osx-64           4.0.1      h1ce6c1d_5  pkgs/main           
clang-dbg_osx-64           4.0.1      h1ce6c1d_7  pkgs/main           
clang-dbg_osx-64           4.0.1      h1ce6c1d_8  pkgs/main           
clang_osx-64              4.0.1               9  pkgs/main           
clang_osx-64              4.0.1     h1ce6c1d_10  pkgs/main           
clang_osx-64              4.0.1      h1ce6c1d_5  pkgs/main           
clang_osx-64              4.0.1      h1ce6c1d_7  pkgs/main           
clang_osx-64              4.0.1      h1ce6c1d_8  pkgs/main           
clang_osx-64              4.0.1      h5d4359d_1  pkgs/main           
clang_osx-64              4.0.1      h60dbf0d_2  pkgs/main           
clang_osx-64              4.0.1      h6c06975_3  pkgs/main           
clang_osx-64              4.0.1      h9d74b7d_5  pkgs/main           
clang_osx-64              4.0.1      hd786faf_0  pkgs/main           
clangxx-dbg_osx-64           4.0.1               5  pkgs/main           
clangxx-dbg_osx-64           4.0.1               7  pkgs/main           
clangxx-dbg_osx-64           4.0.1      h22b1bf0_8  pkgs/main           
clangxx_osx-64            4.0.1               5  pkgs/main           
clangxx_osx-64            4.0.1               7  pkgs/main           
clangxx_osx-64            4.0.1               9  pkgs/main           
clangxx_osx-64            4.0.1      h04dcba0_2  pkgs/main           
clangxx_osx-64            4.0.1     h22b1bf0_10  pkgs/main           
clangxx_osx-64            4.0.1      h22b1bf0_8  pkgs/main           
clangxx_osx-64            4.0.1      h22b6db4_0  pkgs/main           
clangxx_osx-64            4.0.1      h557b0ca_5  pkgs/main           
clangxx_osx-64            4.0.1      h633f504_1  pkgs/main           
clangxx_osx-64            4.0.1      hbbb0622_3  pkgs/main           
gfortran_osx-64           4.8.5      h044dcb3_2  pkgs/main           
gfortran_osx-64           4.8.5      h22b1bf0_3  pkgs/main           
gfortran_osx-64           4.8.5      h878a615_0  pkgs/main           
gfortran_osx-64           4.8.5      hd9f5454_1  pkgs/main           
Loading channels: done
# Name                  Version           Build  Channel             
intel-fortran_win-64          13.1.6      h09e79d1_0  pkgs/main           
intel-fortran_win-64          16.0.4      hc0ab4e9_0  pkgs/main           
intel-fortran_win-64        2017.0.4      h739a657_0  pkgs/main           
vs2008_win-64      9.00.30729.1      hb7fc0b5_0  pkgs/main           
vs2008_win-64      9.00.30729.1      hb7fc0b5_1  pkgs/main           
vs2015_win-64        14.0.25123      h17c34da_3  pkgs/main           
vs2015_win-64        14.0.25123      hf5d5b87_1  pkgs/main           
vs2015_win-64        14.0.25123      hf5d5b87_2  pkgs/main           
vs2017_win-64            15.5.2               0  pkgs/main           
vs2017_win-64            15.5.2               1  pkgs/main           
```
</details>

Reply:
If I remove the `anaconda` channel, and try building this recipe:

```yaml
package:
  name: one
  version: 0.1
requirements:
  build:
    - {{ compiler('c') }}
  host:
    - python
  run:
    - python
```

I get the error `conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform linux-64: {"libgcc-ng[version='>=4.9']"}
`. However if I re-enabling the `anaconda` channel lets the recipe build. I must be doing something wrong, but not sure what.

Reply:
It was, but looking closer I found the problem. 

Due to a longstanding issue with the solver hanging on so many duplicated R packages across channels, I had edited my `~/.condarc` according to this as-yet-unresolved issue: https://github.com/conda/conda/issues/5536:

```yaml
default_channels:
  - https://repo.continuum.io/pkgs/free
```

Changing those lines to:

```yaml
default_channels:
  - https://repo.continuum.io/pkgs/free
  - https://repo.continuum.io/pkgs/main
```

solves the problem. However, since I run everything in the separate "bootstrapped" bioconda-utils miniconda installation, I guess I just learned that my home directory's `.condarc` gets merged into any conda env I use. Good to know.

Reply:
Alternatively, you could also try:
```
conda config --add pinned_packages 'r-base >=3.4'
```
, xref https://github.com/conda/conda/issues/5536#issuecomment-386935346.

----
`~/.condarc` (and others, see: https://github.com/conda/conda/blob/4.5.2/conda/base/constants.py#L25-L43) are pesky little beasts that I always (kind of) deactivate via
```
mkdir -p ~/.conda/condarc.d
touch ~/.conda/condarc ~/.conda/.condarc ~/.condarc
chmod a-w ~/.conda/condarc.d ~/.conda/condarc ~/.conda/.condarc ~/.condarc
```
before `conda` gets a chance to create them.
Sure, they are helpful for end users in some cases, but I find to use `conda config --env` and `conda config --system` much less headache-inducing when operating multiple Conda installations...